plutus-core-spec/plutus-core-specification.tex

```diff
@@ -1033,6 +1027,99 @@ \section{Examples}
+immediately.  As we will see below, \texttt{delay} and \texttt{force}
+are used to preserve the effects of type abstraction and
+instantiation.
+
+As with typed Plutus Core, we can show that values (terms which can
+not be reduced) have a particular syntactic form: see
+Figure~\ref{fig:untyped-values}.
+
+%% The introduction of \texttt{delay} and \texttt{force} may be
+%% controversial since $\delay{M}$ can be implemented as $\lamU{u}{M}$
+%% and $\force{M}$ as $\appU{M}{\lamU{x}{x}}$.  However, doing this
+%% has a cost in code size.  Experiments show that
+%% adding \texttt{delay} and \texttt{force} increases the size of
+%% compressed validators by about 10\%, but using $\lamU{u}{M}$ and
+%% $\appU{M}{\lamU{x}{x}}$ increases sizes by about 20\%.  A third
+%% option is to introduce a global variable $u$ of type \texttt{unit}
+%% and replace $\force{M}$ with $\appU{M}{u}$: this only increases
+%% sizes by about 12\%, but complicates specification because programs
+%% have to be wrapped in an expression binding $u$ (and maybe the CEK
+%% machine has to start with an environment binding $u$ to
+%% $\lamU{x}{x}$).  Another option would be to extend the grammar with
+%% a unit term, but then it would be tempting to add terms
+%% for \texttt{true}, \texttt{false}, and so on.  It's possible that
+%% in actual validator code (especially after type
+%% normalisation), \texttt{delay} and \texttt{force} will typically
+%% appear in expressions of the form $\force{\delay{e}}$, which could
+%% immediately be replaced by $e$ alone, and then using $\lamU{x}{x}$
+%% might not cause so much overhead. I'll look into this.  Later:
+%% nope.  I looked at the on-chain validators for the use cases and
+%% this only happens about 10\% of the time.
+
+\subsection{Type erasure}
+Figure~\ref{fig:type-erasure} defines a type erasure operation which
+converts a typed Plutus Core term $M$ into an untyped one
+$\erase{M}$ (objects from the typed grammar on the left of $=$,
+untyped on the right).  %\footnotemark[4].
+We extend this to sequences of
+terms in the obvious way: given $M^* = M_1 \ldots M_n$ we define
+$\erase{M^*} = \erase{M_1} \ldots \erase{M_n}$.  For a typed program
+$P$ we use $\erase{P}$ to denote the untyped program with the same
+version whose body is the erasure of the body of $P$. The erasure operation
+preserves semantics in an appropriate sense.  It can be shown that if $P$
+is a well-typed Plutus Core program then $P$ terminates if 
+and only if $\erase{P}$ does, and that $P$ returns the value $v$
+if and only if $\erase{P}$ returns the value $\erase{v}$.
+
+%% I'm not wedded to this particular notation for erasure,
+%% but I think it's unlikely that it'll confuse people. Other
+%% possiblities: $\left|M\right|$, $\left\lVert{M}\right\rVert$,
+%% $\overline{M}$ (that looks pretty bad), etc. I'll consult some experts to
+%% find out if there's a standard notation.  (Later: apparently not.)
```

Comment:
Seems fine. Random paper I looked at (https://ecommons.cornell.edu/bitstream/handle/1813/7375/98-1721.pdf?sequence=1) uses `M^o` or similar, which isn't very nice either.

If people complain it looks like evaluation, we can just say it *is* a kind of evaluation :D

Reply:
That's my thinking too!